bin/underreact.js

```diff
@@ -120,87 +111,69 @@ function defineServeStatic(y) {
-      normalize: true,
-      default: '.'
-    })
-    .help();
-}
-
-function runWriteBabelrc(argv) {
-  try {
-    writeBabelrc(getConfig('write-babelrc', argv), argv.output, argv.env);
-  } catch (error) {
-    errorOut(error);
-  }
+  getConfig('serve-static', argv)
+    .then(urc => serveStatic(urc))
+    .catch(errorOut);
 }
 
 function getConfig(command, argv) {
   const configIsNotSpecified = argv.config === undefined;
   let configPath;
+
   if (configIsNotSpecified) {
-    configPath = path.join(process.cwd(), 'underreact.config.js');
+    configPath = path.join(process.cwd(), DEFAULT_CONFIG_NAME);
   } else {
-    configPath = path.isAbsolute(argv.config)
-      ? argv.config
-      : path.join(process.cwd(), argv.config);
-  }
-  let production = true;
-  if (command === 'build') {
-    production = !argv.debug;
-  } else if (command === 'start') {
-    production = argv.production;
-  } else if (command === 'write-babelrc' && argv.env === 'prod') {
-    production = true;
+    configPath = path.resolve(argv.config);
   }
 
-  // This object is passed as the argument to the config module, if it's a
-  // function.
-  const configModuleContext = {
-    webpack,
-    command,
-    production,
-    argv,
-    configDir: path.dirname(configPath),
-    stats: argv.stats,
-    port: argv.port
-  };
-
-  const defaultConfig = getDefaultConfig(configModuleContext);
-
-  let userConfig = {};
-  // If the user didn't use --config, it's fine if the default does not exist.
-  // But if they did specify a path, tell them if it doesn't work.
-  try {
-    const configModule = require(configPath);
-    userConfig =
-      typeof configModule === 'function'
-        ? configModule(configModuleContext)
-        : configModule;
-  } catch (error) {
-    if (!configIsNotSpecified) {
-      logger.error(
-        `Failed to load configuration module from ${chalk.underline(
+  return userConfigReader(configPath)
+    .catch(() => {
+      // If the user didn't use --config, it's fine if the underreact.config.js
+      // does not exist at the default path.
+      if (configIsNotSpecified) {
+        return () => {};
+      }
+      // But if they did specify a path, tell them it didn't work.
+      throw new Error(
+        `Failed to load underreact configuration module from ${chalk.underline(
           configPath
         )}`
       );
-    }
-  }
+    })
+    .then(getUserConfig => {
+      let production = true;
+      if (command === 'build') {
+        production = !argv.debug;
+      } else if (command === 'start') {
+        production = argv.production;
+      } else if (argv.env === 'prod') {
+        production = true;
+      }
+      // This object is passed as the argument to the config module, if it's a
+      // function.
+      const configModuleContext = {
+        webpack,
+        command,
+        production,
+        argv,
+        configDir: path.dirname(configPath),
+        stats: argv.stats,
+        port: argv.port
```

Comment:
Hm. The internal one should need stats and port, either, since it can get that info off `argv`, right?

I'd definitely favor a clean, tailored public API over this kind of consistency. If that means we need change the implementation details, then I think that's worth it.

Reply:
I just fixed it in the latest commit 🔨